Add initial CODEOWNERS

ddbeck opened the pull request:
The new governance doc (#3668) calls for a CODEOWNERS file: 

> The owners are also listed as top-level owners in GitHub’s code owner file.

This PR adds this, though we might might want to fill it out a bit more, unless we want every PR to request the owners' reviews

vinyldarkscratch commented:
Basically, what we’re looking for then, is the following:

```
webextensions/ @irenesmith
!*.json @Elchi3
```

Based upon what I’ve been reading, CODEOWNERS is a similar syntax to .gitignore files, so the ! should work, theoretically.  It seems to choose the last match, so any non-JSON files in webextensions/ will not be assigned to Irene and only assigned to Elchi3.

ddbeck commented:
OK, I've updated the `CODEOWNERS` file to reflect @vinyldarkscratch's suggested syntax and added a comment referring to the governance doc. Rather than being a circular reference, I dropped the code owners line from the governance doc.

vinyldarkscratch commented:
Hmm, it looks like it may not be functioning as expected.  I reviewed a PR that altered the test files (JS files), and didn’t automatically add @Elchi3 for review.  More research may be needed.

ddbeck commented:
Making a note here: PR #3933 is addressing this.
